src/gui/tray/ActivityItemContent.qml

```diff
@@ -19,19 +20,57 @@ RowLayout {
     signal dismissButtonClicked()
     signal shareButtonClicked()
 
-    spacing: 10
-
-    Image {
-        id: activityIcon
+    spacing: Style.trayHorizontalMargin
 
+    Item {
         Layout.alignment: Qt.AlignVCenter | Qt.AlignHCenter
-        Layout.preferredWidth: 32
-        Layout.preferredHeight: 32
+        Layout.preferredWidth: Style.trayListItemIconSize
+        Layout.preferredHeight: Style.trayListItemIconSize
+
+        Image {
+            id: thumbnailImage
+            width: model.thumbnail.isMimeTypeIcon ? parent.width * 0.85 : parent.width * 0.8
+            height: model.thumbnail.isMimeTypeIcon ? parent.height * 0.85 : parent.height * 0.8
+            anchors.verticalCenter: parent.verticalCenter
+            anchors.left: parent.left
+            cache: true
+            source: model.thumbnail.source
```

Comment:
`model.thumbnail` can be returned undefined in that case every usage of it in this component will generate a warning
can you check if you can use the same logic that is used in `activityIcon` ?

Reply:
Good catch, I added a check for `model.thumbnail` not being undefined to set the property